mozilla_aws_cli/sts_conn.py

```diff
@@ -43,20 +48,31 @@ def get_credentials(bearer_token, id_token_dict, role_arn):
 
             # Call the STS API
             resp = requests.get(url=sts_url, params=parameters)
+            try:
+                root = ElementTree.fromstring(resp.content)
+            except ElementTree.ParseError as e:
+                raise MalformedResponseWarning('Unable to parse XML response to AssumeRoleWithWebIdentity call')
             if resp.status_code != requests.codes.ok:
-                if 'The requested DurationSeconds exceeds the MaxSessionDuration set for this role' in resp.text:
+                error = dict(
+                    [(x.tag.split('}', 1)[-1], x.text) for x in root.find(
+                        './sts:Error',
+                        {'sts': 'https://sts.amazonaws.com/doc/2011-06-15/'})])
+                logger.error(
+                    'AWS STS Call failed {status} {Type} {Code} : {Message}'.format(
+                        status=resp.status_code, **error))
+                if 'The requested DurationSeconds exceeds the MaxSessionDuration set for this role' in error['Message']:
```

Comment:
I assume there is no other way to check this? This seems pretty fragile.

Reply:
The AWS docs don't show anything. Lemme see if I can force the case to happen and see if there's anything more deterministic to match on.

Reply:
Ya the full error is
```
{'Type': 'Sender', 'Code': 'ValidationError', 'Message': 'The requested DurationSeconds exceeds the MaxSessionDuration set for this role.'}
```
Such that `ValidationError` is ambiguous so we have to use the `Message` value

Reply:
I'll add that `Code` check as well to be explicit

Reply:
Can we just key off of `Code` == `ValidationError`? I'm sure it's not a 1:1 mapping, but if it's the only thing in our code that triggers it then it seems fine?

Reply:
Fixed this in 44d6ca61a603e510a17d7b703d09c8c9a5549657